### PR TITLE
Bucket Sort implementation

### PR DESCRIPTION
**Pull request by tuqqu**
## Description

Bucket Sort implementation.
## How Has This Been Tested?

Tests are done similarly to the existing tests for sorting algorithms.

### CODE REVIEW

**Comment by tuqqu** [comment]
@doganoo Hi! Is there anything that needs to be done before this PR is merged?

File: src/Algorithm/Sorting/BucketSort.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
+<?php
+declare(strict_types=1);
+/**
+ * MIT License
+ *
+ * Copyright (c) 2018 Dogan Ucar
```
Comment (doganoo) [inline src/Algorithm/Sorting/BucketSort.php:6]: could you please add your own name here?

**Review by doganoo — CHANGES_REQUESTED** [review]
@tuqqu Thank you for contributing to PHPAlgorithms! Please excuse my late reply, I was a bit busy with projects :-)

There is just a simple small change - could you please add your own name to the file header?

**Comment by tuqqu** [comment]
No worries! Added an author tag to the file.

**Comment by doganoo** [comment]
👍👍 thanks. Will merge the PR and tag ASAP

**Pull request merged**